src/java.base/share/classes/sun/security/provider/SunEntries.java

```diff
@@ -115,218 +118,266 @@ public static List<String> createAliasesWithOid(String ... oids) {
+                    null, attrs);
+        } else {
+            /*
+            * SecureRandom engines
+            */
+            attrs.put("ThreadSafe", "true");
+            if (NativePRNG.isAvailable()) {
+                add(p, "SecureRandom", "NativePRNG",
+                        "sun.security.provider.NativePRNG",
+                        null, attrs);
+            }
+            if (NativePRNG.Blocking.isAvailable()) {
+                add(p, "SecureRandom", "NativePRNGBlocking",
+                        "sun.security.provider.NativePRNG$Blocking", null, attrs);
+            }
+            if (NativePRNG.NonBlocking.isAvailable()) {
+                add(p, "SecureRandom", "NativePRNGNonBlocking",
+                        "sun.security.provider.NativePRNG$NonBlocking", null, attrs);
+            }
+            attrs.put("ImplementedIn", "Software");
+            add(p, "SecureRandom", "DRBG", "sun.security.provider.DRBG",
                 null, attrs);
+            add(p, "SecureRandom", "SHA1PRNG",
+                    "sun.security.provider.SecureRandom", null, attrs);
 
-        /*
-         * Policy
-         */
-        add(p, "Policy", "JavaPolicy", "sun.security.provider.PolicySpiFile",
-                null, null);
-
-        /*
-         * Configuration
-         */
-        add(p, "Configuration", "JavaLoginConfig",
-                "sun.security.provider.ConfigFile$Spi", null, null);
-
-        /*
-         * CertPathBuilder and CertPathValidator
-         */
-        attrs.clear();
-        attrs.put("ValidationAlgorithm", "RFC5280");
-        attrs.put("ImplementedIn", "Software");
-
-        add(p, "CertPathBuilder", "PKIX",
-                "sun.security.provider.certpath.SunCertPathBuilder",
-                null, attrs);
-        add(p, "CertPathValidator", "PKIX",
-                "sun.security.provider.certpath.PKIXCertPathValidator",
-                null, attrs);
+            /*
+            * Signature engines
+            */
+            attrs.clear();
+            String dsaKeyClasses = "java.security.interfaces.DSAPublicKey" +
+                    "|java.security.interfaces.DSAPrivateKey";
+            attrs.put("SupportedKeyClasses", dsaKeyClasses);
+            attrs.put("ImplementedIn", "Software");
+
+            attrs.put("KeySize", "1024"); // for NONE and SHA1 DSA signatures
+
+            add(p, "Signature", "SHA1withDSA",
+                    "sun.security.provider.DSA$SHA1withDSA",
+                    createAliasesWithOid("1.2.840.10040.4.3", "DSA", "DSS",
+                        "SHA/DSA", "SHA-1/DSA", "SHA1/DSA", "SHAwithDSA",
+                        "DSAWithSHA1", "1.3.14.3.2.13", "1.3.14.3.2.27"), attrs);
+            add(p, "Signature", "NONEwithDSA", "sun.security.provider.DSA$RawDSA",
+                    createAliases("RawDSA"), attrs);
+
+            attrs.put("KeySize", "2048"); // for SHA224 and SHA256 DSA signatures
+
+            add(p, "Signature", "SHA1withDSA",
```

Comment:
Please explain why this should change for non-FIPS mode.

Reply:
Can you please answer this query?

Reply:
That specific code disables non-certified FIPS cryptographic algorithms in the SUN security provider once the JDK is placed in FIPS mode. 

The reason we don't disable the entire security provider is that there are some needed services in the SUN provider that are not cryptographic in nature and are needed - for example, the security policy service provider and some certificate manipulation services.

Reply:
This code applies only if `FIPSConfigurator.enableFips()` answers `false`, so the JDK is not in FIPS mode, or do I misunderstand something?

Reply:
the code is 

```
if (isFips) {
.. add only specific services that are needed in FIPS mode ...
} else {
.. add everything that SUN provider supports, including non FIPS mode ....
}
```

so when in isFIPS==true, only those specific services are activated and everything else is not.

Reply:
i dont believe it is changed - just the diff got messed up

Reply:
It did change, from
```
        add(p, "Signature", "SHA224withDSA",
```
to
```
            add(p, "Signature", "SHA1withDSA",
```
The indentation change I expected, but not the change from `"SHA224withDSA"` to `"SHA1withDSA"`.
Looking at https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/520/files?w=1 should clarify that.

Reply:
@taoliult please fix